### PR TITLE
Non-null filtering, because sometimes insn's are null and transformer…

### PR DESCRIPTION
**Pull request by Animowany**
… catches error.

### CODE REVIEW

File: deobfuscator-transformers/src/main/java/uwu/narumi/deobfuscator/core/other/impl/universal/number/CompareInstructionsTransformer.java [hunk 1]
```diff
@@ -19,7 +20,8 @@ public class CompareInstructionsTransformer extends FramedInstructionsTransforme
   @Override
   protected Stream<AbstractInsnNode> getInstructionsStream(Stream<AbstractInsnNode> stream) {
     return stream
-        .filter(insn -> AsmMathHelper.isMathCompare(insn.getOpcode()));
+            .filter(Objects::nonNull)
```
Comment (EpicPlayerA10) [inline deobfuscator-transformers/src/main/java/uwu/narumi/deobfuscator/core/other/impl/universal/number/CompareInstructionsTransformer.java:23]: You don't need to check here if it is null as you already check it in `FramedInstructionsTransformer`

File: deobfuscator-transformers/src/main/java/uwu/narumi/deobfuscator/core/other/impl/universal/number/MathOperationsTransformer.java [hunk 2]
```diff
@@ -19,7 +20,8 @@ public class MathOperationsTransformer extends FramedInstructionsTransformer {
   @Override
   protected Stream<AbstractInsnNode> getInstructionsStream(Stream<AbstractInsnNode> stream) {
     return stream
-        .filter(insn -> AsmMathHelper.isMathOperation(insn.getOpcode()));
+            .filter(Objects::nonNull)
```
Comment (EpicPlayerA10) [inline deobfuscator-transformers/src/main/java/uwu/narumi/deobfuscator/core/other/impl/universal/number/MathOperationsTransformer.java:23]: same here

File: deobfuscator-transformers/src/main/java/uwu/narumi/deobfuscator/core/other/impl/universal/number/NumberCastsTransformer.java [hunk 3]
```diff
@@ -18,7 +19,8 @@ public class NumberCastsTransformer extends FramedInstructionsTransformer {
   @Override
   protected Stream<AbstractInsnNode> getInstructionsStream(Stream<AbstractInsnNode> stream) {
     return stream
-        .filter(insn -> AsmMathHelper.isNumberCast(insn.getOpcode()));
+            .filter(Objects::nonNull)
```
Comment (EpicPlayerA10) [inline deobfuscator-transformers/src/main/java/uwu/narumi/deobfuscator/core/other/impl/universal/number/NumberCastsTransformer.java:22]: same here

**Review by EpicPlayerA10 — CHANGES_REQUESTED** [review]
If instruction is null then some transformer is badly designed. But if you really want i've requested some small changes.

**Comment by EpicPlayerA10** [comment]
And for future reference, don't commit from a master branch, but create a new fresh branch for your PR. This will make it easier for you in your future PRs to any open-source repository.

**Comment by Animowany** [comment]
You wrote universalnumbertransformer, and i occured many nullpointers, so just inserted a check and now everything is fine.

**Comment by Animowany** [comment]
> > You wrote universalnumbertransformer, and i occured many nullpointers, so just inserted a check and now everything is fine.
> 
> Hmm. That's unexpected behaviour. Maybe i do some sanity checks after each transformer if any transformer didn't accidentally break.

im now deobfuscating pretty heavily obfuscated jar, so maybe it has some  as null presented insn's if its even possible

**Comment by EpicPlayerA10** [comment]
> > > You wrote universalnumbertransformer, and i occured many nullpointers, so just inserted a check and now everything is fine.
> > 
> > 
> > Hmm. That's unexpected behaviour. Maybe i do some sanity checks after each transformer if any transformer didn't accidentally break.
> 
> im now deobfuscating pretty heavily obfuscated jar, so maybe it has some as null presented insn's if its even possible

If jar is valid and it is running well, then there shouldn't be any null instructions. Some transformer might broke it.

**Comment by Janmm14** [comment]
i neverhad to check an instruction being null

**Comment by narumii** [comment]
> > > > You wrote universalnumbertransformer, and i occured many nullpointers, so just inserted a check and now everything is fine.
> > > 
> > > 
> > > Hmm. That's unexpected behaviour. Maybe i do some sanity checks after each transformer if any transformer didn't accidentally break.
> > 
> > 
> > im now deobfuscating pretty heavily obfuscated jar, so maybe it has some as null presented insn's if its even possible
> 
> If jar is valid and it is running well, then there shouldn't be any null instructions. Some transformer might broke it.

Perhaps better solution would be attaching the jar in question as a test suite?
Null insn should not happen, but at this point i would not be suprised since it is asm.

**Comment by EpicPlayerA10** [comment]
@Animowany Your issue should be partially resolved by https://github.com/narumii/Deobfuscator/commit/521c096282f43fdfff1b069d7f8f156ee4fc5ff9 . Now the deobfuscator will throw an error which transformer did something bad in instructions list, instead of acting this weird behaviour with replacing all instructions with nulls. By this change you can then find the root cause of an error very easily.

**Pull request closed without merging**